lib/vsc/administration/slurm/scontrol.py

```diff
@@ -0,0 +1,279 @@
+#
+# Copyright 2022-2022 Ghent University
+#
+# This file is part of vsc-administration,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-administration
+#
+# All rights reserved.
+#
+"""
+scontrol commands
+"""
+import logging
+import re
+import shlex
+
+from enum import Enum
+
+from vsc.accountpage.wrappers import mkNamedTupleInstance
+from vsc.utils.missing import namedtuple_with_defaults
+from vsc.utils.run import asyncloop
+
+
+SLURM_SCONTROL = "/usr/bin/scontrol"
+
+SLURM_SCONTROL_CONFIG_REGEX = re.compile("^(.*\S)\s+=\s+(\S.*)$")
+
+
+LICENSE_RESERVATION_PREFIX = 'external_license_'
+
+class ScontrolTypes(Enum):
+    reservation = "reservation"
+    license = "license"
+    config = "config"
+    partition = "partition"
+
+
+ScontrolReservationFields = [
```

Comment:
Ideally, these things are versioned. They might change and we should then only add the new ones and use the correct version key.